### ISSUE TITLE
store ipython profile name in RE metadata

### ISSUE THREAD

**Issue by prjemian** [NSLS-II/bluesky#1049]
## Expected Behavior
Wouldn't it be nice if ...

the metadata included the name of the ipython profile
for diagnostic purposes, of course

## Current Behavior
doesn't do this

## Possible Solution
I've got code

## Steps to Reproduce (for bugs)
enhancement

## Context
supporting users for when they do not know what went wrong

## Your Environment
any

**Comment by prjemian** [comment]
Where to put this single function code?  bluesky.utils?

**Comment by prjemian** [comment]
This code: https://github.com/APS-2BM-MIC/ipython-user2bmb/blob/master/profile_2bmb/startup/60-metadata.py#L8-L27

**Comment by danielballan** [comment]
Yeah, utils seems like the best place for this.

**Comment by ke-zhang-rd** [comment]
The code below could possibly be a alternative for your request.

from IPython import get_ipython
RE.md['ipython_profile'] = get_ipython().profile  # returns string profile name

The thing is `__file__` will make us maintain this for each beamline independently which is not what we want. The method above will be available on document later.

What do you think. @prjemian

**Comment by prjemian** [comment]
+1 attaboy!

**Comment by mrakitin** [comment]
Reopening. Will be closed automatically once #1128 is merged.

**Comment by tacaswell** [comment]
To re-iterate what @ke-zhang-rd said

```python
from IPython import get_ipython
RE.md['ipython_profile'] = get_ipython().profile
```

is the solution for how to do this.
